class-search-swagger.yaml

```diff
@@ -0,0 +1,275 @@
+swagger: '2.0'
+
+info:
+  version: "0.0.1"
+  title: Class Search API
+
+# Describe your paths here
+paths:
+  # This is a path endpoint. Change it.
+  /class-search:
+    # This is a HTTP operation
+    get:
+      tags:
+       - classes
+       - search
+      # Describe this verb here. Note: you can use markdown
+      description: "Gets all `class` objects for a term that are available for registration."
+      operationId: getClasses
+      # This is array of GET operation parameters:
+      produces:
+        - application/json
+      parameters:
+        - in: query
+          name: term
+          description: "todo"
+          required: true
+          type: string
+        - in: query
+          name: subject
+          description: "todo"
+          required: false
+          type: string
+        - in: query
+          name: courseNumber
+          description: "todo"
+          required: false
+          type: string
+        - in: query
+          name: q
+          description: "Searches class title, description, and subject"
+          required: false
+          type: string
+        - in: query
+          name: page[size]
+          description: "Number of results to return. Used in pagination. Default value is 10."
+          required: false
+          type: string
+        - in: query
+          name: page[number]
+          description: "Page number of results. Used to paginate through results."
+          required: false
+          type: string
+        
+          
+      # Expected responses for this operation:
+      responses:
+        # Response code
+        "200":
+          description: "Successful response"
+          schema:
+            $ref: "#/definitions/ResultObjects"
+        "404":
+          description: "Not Found"
+          schema:
+            $ref: "#/definitions/Error"
+        "500":
+          description: "Internal Server Error"
+          schema:
+            $ref: "#/definitions/Errors"
+      security:
+        - locations_api_key: []
```

Comment:
Good point. I used to specify to import the swagger and security information into the API gateway, but later on found out that the gateway didn't support this part of the swagger spec :(